src/main/java/io/cryostat/net/web/http/api/beta/ProbeTemplateUploadHandler.java

```diff
@@ -0,0 +1,160 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta;
+
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.core.agent.LocalProbeTemplateService;
+import io.cryostat.core.agent.ProbeValidationException;
+import io.cryostat.core.log.Logger;
+import io.cryostat.core.sys.FileSystem;
+import io.cryostat.messaging.notifications.NotificationFactory;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.FileUpload;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+class ProbeTemplateUploadHandler extends AbstractV2RequestHandler<Void> {
+
+    static final String PATH = "probes/:probetemplateName";
+
+    private static Logger logger;
+    private final NotificationFactory notificationFactory;
+    private final LocalProbeTemplateService probeTemplateService;
+    private final FileSystem fs;
+    private static final String NOTIFICATION_CATEGORY = "ProbeTemplateUploaded";
+
+    @Inject
+    ProbeTemplateUploadHandler(
+            AuthManager auth,
+            NotificationFactory notificationFactory,
+            LocalProbeTemplateService probeTemplateService,
+            Logger logger,
+            FileSystem fs,
+            Gson gson) {
+        super(auth, gson);
+        this.notificationFactory = notificationFactory;
+        this.logger = logger;
+        this.probeTemplateService = probeTemplateService;
+        this.fs = fs;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.CREATE_PROBE_TEMPLATE);
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse handle(RequestParameters requestParams) throws Exception {
+        try {
+            for (FileUpload u : requestParams.getFileUploads()) {
+                String templateName = requestParams.getPathParams().get("probetemplateName");
```

Comment:
The `probetemplateName` path param is effectively a constant for the duration of this HTTP request, but it's used here within the loop for the name of each template added corresponding to each file upload. So, if I make a POST request to this handler that contains 2 or more file uploads, I will get an exception thrown by the template service:

https://github.com/cryostatio/cryostat-core/blob/af317dfbcd84def96e0a15d18c74015e19265fbf/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java#L92

This seems a bit non-obvious. The way this is typically done in other API handlers is to require the `FileUpload` to have a specific identifying name, like `template`: https://github.com/cryostatio/cryostat/blob/main/src/main/java/io/cryostat/net/web/http/api/v1/TemplatesPostHandler.java#L120

This way, the client should be aware that only one single file upload is expected by this handler. If they send it with the wrong name, or send other files along with other names, then those will be ignored and deleted.

Reply:
Makes sense, I've added a check for the upload name along the same lines as the linked example, thanks!